poppy/wfe.py

```diff
@@ -323,3 +324,32 @@ def get_opd(self, wave, units='meters'):
         if units == 'waves':
             opd /= wave.wavelength.to(u.meter).value
         return opd
+
+
+class StatsticalPSDWFE(WavefrontError):
+    """
+    Statistical PSF WFE class from power law for optical noise.
+
+    Input WFE in radians RMS.
+    """
+
+    @utils.quantity_input(wfe=u.rad)
+    def __init__(self, name='PSD WFE', index=3.0, wfe=0.25, **kwargs):
+
+        super(WavefrontError, self).__init__(name=name, **kwargs)
+        self.index = index
+        self.wfe_rad = wfe
+
+    def get_opd(self, wave, units='meters'):
+
+        y, x = self.get_coordinates(wave)
+        rho, theta = _wave_y_x_to_rho_theta(y, x, self.radius.to(u.meter).value)
+        psd = np.power(rho, -self.index)
+        psd = psd / np.sum(psd) * np.square(self.wfe_rad / wave.wavelength.to(u.meter).value)
+
+        rndm_phase = np.random.random((len(y), len(x)))
```

Comment:
Gauss distributed samples would be better here; plus adding a seed for the random number generator to be able to regenerate exactly the same phase screens.

Reply:
Agreed. Seed could be a parameter passed in to the init function of the class.